fileio.c

```diff
@@ -423,7 +423,7 @@ char *getpathforfileindir(char const *dir, char const *filename)
     int		m, n;
 
     m = strlen(filename);
-    if (!dir || !*dir || strchr(filename, DIRSEP_CHAR)) {
+    if (!dir || !*dir) {
```

Comment:
Does this still work properly with absolute paths?  I assume that's what the original check was for...

Reply:
No, it doesn't, but I think I checked all usages of the function to never use absolute paths and only relative ones.